src/app/shared/scroll-with-loading/scroll-with-loading.directive.ts

```diff
@@ -0,0 +1,28 @@
+import {Directive, output} from '@angular/core';
+
+@Directive({
+    selector: '[appScrollWithLoading]',
+    standalone: true,
+    host: {
+        '(scroll)': 'handleScroll($event)',
+    },
+})
+export class ScrollWithLoadingDirective {
+    private readonly borderOffset = 100;
+    private scrollTopLastValue = 0;
+    readonly loadNextData = output();
+
+    private handleScroll(event: Event) {
+        if (event.target instanceof HTMLElement) {
```

Comment:
Тут несколько рекомендаций дам по хорошим практикам:
1. Ранний ретерн. Он поможет уменьшить вложенность и улучшить читаемость в особенно сложных кейсах. Данный подход в целом помогает сконцентрироваться на главной задаче функции, ведь ранний ретерн по сути - исключение
2. Деструктуризация. Удобно создавать переменные выполняя деструктуризацию объектов, рекомендую взять на вооружение. Вижу что ниже ты ей воспользовался, было бы круто и в аргумент занести

```suggestion
    private handleScroll({target}: Event) {
        if (!target instanceof HTMLElement) {
             return;
         }
```

Reply:
Обновил.